src/rust/engine/fs/store/src/lib.rs

```diff
@@ -1987,6 +2103,18 @@ mod remote {
             .to_boxed(),
           }
         })
+        .then(move |future| {
+          let time_span = super::TimeSpan::since(&start_time);
```

Comment:
Since this code is repeated in a few places, a helper function would be useful.

Since this is green, I'm going to go ahead and merge, but it would be good to follow up to dedupe this sooner rather than later.